WordPress/Classes/Stores/StatsInsightsStore.swift

```diff
@@ -1059,8 +1070,19 @@ private extension StatsInsightsStore {
         NotificationCenter.default.addObserver(forName: .WPAccountDefaultWordPressComAccountChanged,
                                                object: nil,
                                                queue: nil) { notification in
-            HomeWidgetTodayData.delete()
+            if !AccountHelper.isLoggedIn {
+                HomeWidgetTodayData.delete()
+            }
+            UserDefaults(suiteName: WPAppGroupName)?.setValue(AccountHelper.isLoggedIn, forKey: WPStatsHomeWidgetsUserDefaultsLoggedIndKey)
```

Comment:
Would probably be good to have a helper method for setting and getting this value, but we can do that at another time.